### PR TITLE
systemd: implement restart as a single systemctl restart call

### PR DESCRIPTION
**Pull request by bboozzoo**
Update related tests.

Related to #10164. Let's see if calling systemctl restart fails the test suite.

### CODE REVIEW

**Comment by anonymouse64** [comment]
this needs to be rebased on master or a master merge in order to pass spread tests, but before I retried them, they looked like standard failures we see from the store or otherwise known flakiness, so this may be okay to do, but we should make this a conscious decision and merge very early to get as much testing as possible before it releases, hence 2.52 timeline

**Comment by bboozzoo** [comment]
Probably similar to https://github.com/snapcore/snapd/pull/10166, but haven't looked in great details. Closing for now, can always come back to this branch later.

**Comment by bboozzoo** [comment]
Rebased on top of current master and force pushed. Needs a proper review now.

File: systemd/systemd.go [hunk 1]
```diff
@@ -834,10 +834,9 @@ func (s *systemd) Restart(serviceName string, timeout time.Duration) error {
 	if s.mode == GlobalUserMode {
 		panic("cannot call restart with GlobalUserMode")
 	}
-	if err := s.Stop(serviceName, timeout); err != nil {
-		return err
-	}
-	return s.Start(serviceName)
+	// XXX: timeout is ignored
```
Comment (mardy) [inline systemd/systemd.go:837]: I'm not sure what the "`XXX`" means in the context of a comment; is it something you plan to do before removing the "draft" flag from the PR?

Comment (bboozzoo) [reply systemd/systemd.go:837]: No, it's about timeout being passed to Restart() as mandated by our systemd interface definition, but getting ignored here.

Comment (mardy) [reply systemd/systemd.go:837]: So, it's just to say that we are aware of the timeout, but that we cannot implement it?

If there is no way to implement the timeout mechanism in this case, maybe we should have a 
```
// TODO: remove unused timeout parameter from the interface
```

instead?

**Comment by bboozzoo** [comment]
Marked this as ready for review. The test suite is happy, so it isn't entirely clear why we had stop/show/start sequence in the first place?

**Review by anonymouse64 — APPROVED** [review]
lgtm, and all the tests seem to be happy, so I think this should be good to go into 2.53, but definitely this is a change we want very widely tested.

**Review by MiguelPires — APPROVED** [review]
LGTM

**Review by anonymouse64 — APPROVED** [review]
still lgtm, I think merging this soon is a good idea since it will probably be a while before 2.54 hits a release channel

**Review by mardy — APPROVED** [review]
LGTM!

**Comment by bboozzoo** [comment]
Superseded by #11314

**Pull request closed without merging**